### PR TITLE
Add FloatCrossFloatTransform

### PR DESCRIPTION
**Pull request by christhetree**
to: @jq 
cc: @deerzq 

Content:
This is just a quick foundation of what I envisioned a float cross float transform might look like (essentially a MoveFloatToString transform followed by a StringCrossFloat transform, but without deleting the original float feature or creating an intermediate string feature which would occur if these two transforms were applied back to back). The ability to specify which keys to cross could be added as well instead of crossing the entire family.

Please let me know what you think / if this is what you need. Once we're on the same page I'll add comments and tests. Thanks!

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    for (Map.Entry<String, Double> entry1 : map1.entrySet()) {
+      String float1Key = entry1.getKey();
+      Double float1Value = entry1.getValue();
+
+      if (float1Value > cap) {
+        float1Value = cap;
+      }
+
+      Double float1Quantized = LinearLogQuantizeTransform.quantize(float1Value, bucket);
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java]: perhaps we should allow the user select the quantization method

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package com.airbnb.aerosolve.core.transforms;
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.util.Util;
+
+import java.util.Map;
+
+import com.typesafe.config.Config;
+
+public class FloatCrossFloatTransform extends Transform {
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java]: more comments about the transform usage

**Comment by jq** [comment]
:+1:  go ahead and add test + comments, for "user select the quantization method" you can pick either add in this pr or next pr.

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java (hunk 1 above)
Comment (christhetree) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java]: I'll add this in another PR.

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java (hunk 2 above)
Comment (christhetree) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java]: Done.

**Comment by christhetree** [comment]
@jq @deerzq Tests and comments added PTAL

File: core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java (hunk 1 above)
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/transforms/FloatCrossFloatTransform.java]: sounds good

**Comment by deerzq** [comment]
LGTM

**Comment by jq** [comment]
:+1:

**Pull request merged**